File: packages/ui/lib/utils/quest-file/genQuestProp.ts
export interface Quantity {
    gathering_tables_qty: number;
    unk0: number;
    area1zones: number;
    area2zones: number;
    area3zones: number;
    area4zones: number;
}

export interface GenQuestProp {
    big_monster_size_multi: number;
    size_range: number;
    mons_stat_table1: number;
    main_rank_points: number;
    kn: number;
    sub_arank_points: number;
    sub_brank_points: number;
    monster_class_id: number;
    // skip 4
    skip1: number[];
    little_mons_stat_table: number;
    // skip 9
    skip2: number[];
    quest_kn0: number;
    // skip 7
    skip3: number[];
    quest_kn1: number;
    quest_kn2: number;
    quest_kn3: number;
    quantity: Quantity,
    unk3: number;
    unk4: number;
    unk5: number;
}

File: src-tauri/src/editor/tests/gen_quest_prop.rs
use crate::editor::{
    file::reader::FileReader,
    quest::offsets::GEN_QUEST_PROP_PRT,
    quest::quest_type_flags::{GenQuestProp, Quantity},
};

#[test]
fn gen_quest_prop() {
    let mut reader = FileReader::from_filename("quest-tests/21085d0.bin").unwrap();

    let expected = GenQuestProp {
        big_monster_size_multi: 102,
        size_range: 3,
        mons_stat_table1: 13,
        main_rank_points: 530,
        kn: 15,
        sub_arank_points: 80,
        sub_brank_points: 80,
        monster_class_id: 1,
        skip1: [1, 1, 0, 0],
        little_mons_stat_table: 13,
        skip2: [0, 0, 1, 0, 0, 0, 57, 0, 3],
        quest_kn0: 1,
        skip3: [0; 7],
        quest_kn1: 2,
        quest_kn2: 3,
        quest_kn3: 56,
        quantity: Quantity {
            gathering_tables_qty: 102,
            unk0: 6,
            area1zones: 11,
            area2zones: 11,
            area3zones: 11,
            area4zones: 11,
        },
        unk3: 0,
        unk4: 0,
        unk5: 0,
    };

    reader.seek_start(GEN_QUEST_PROP_PRT as u64).unwrap();
    let gen_quest_prop = reader.read_struct::<GenQuestProp>().unwrap();

    assert_eq!(gen_quest_prop, expected);
}
